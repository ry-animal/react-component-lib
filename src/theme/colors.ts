import { ThemeColors, ThemeName } from "./theme.interface";

export const lightPaletteColors = {
  PRIMARY_1: "#000000",
  PRIMARY_2: "#5F5F5F",
  PRIMARY_3: "#EEEEEE",

  SECONDARY_1: "#FFF6DA",
  SECONDARY_2: "#FFE8A1",
  SECONDARY_3: "#FFD968",

  SUCCESS_1: "#F2FBF8",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const lightPaletteBase = {
  SURFACE_1: "#F9F9F9",
  SURFACE_2: "#FFFFFF",
  SURFACE_3: "#EEEEEE",
  SURFACE_4: "#DBDBDB",

  TEXT_1: "#151515",
  TEXT_2: "#404040",
  TEXT_3: "#5F5F5F",
  TEXT_4: "#A4A4A4",
};

export const darkPaletteColors = {
  PRIMARY_1: "#F9F9F9",
  PRIMARY_2: "#EEEEEE",
  PRIMARY_3: "#151515",

  SECONDARY_1: "#FFE8A1",
  SECONDARY_2: "#FFD968",
  SECONDARY_3: "#B38402",

  SUCCESS_1: "#19C68C",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const darkPaletteBase = {
  SURFACE_1: "#151515",
  SURFACE_2: "#212322",
  SURFACE_3: "#404040",
  SURFACE_4: "#5F5F5F",

  TEXT_1: "#FFFFFF",
  TEXT_2: "#EEEEEE",
  TEXT_3: "#DBDBDB",
  TEXT_4: "#C3C3C3",
};

export const palletteBase = {
  BLACK: "#000000",
  WHITE: "#ffffff",
};

export const PALETTE_LIGHT: ThemeColors = {
  ...lightPaletteColors,
  ...lightPaletteBase,
  ...palletteBase,
};

export const PALETTE_DARK: ThemeColors = {
  ...darkPaletteColors,
  ...darkPaletteBase,
  ...palletteBase,
};

export function getColorsByTheme(theme: ThemeName): ThemeColors {
  if (theme === ThemeName.DARK) {
    return PALETTE_DARK;
  }

  return PALETTE_LIGHT;
}
