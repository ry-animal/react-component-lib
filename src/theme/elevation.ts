export const ELEVATION_LIGHT = {
  low: "0px 2px 8px RGBA(0, 0, 0, 0.04)",
  medium: "0px 2px 16px RGBA(0, 0, 0, 0.16)",
  high: "0px 2px 24px RGBA(0, 0, 0, 0.24)",
};

export const ELEVATION_DARK = {
  low: "inherit",
  medium: "inherit",
  high: "inherit",
};
