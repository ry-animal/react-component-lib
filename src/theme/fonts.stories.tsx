import * as React from "react";
import { lightPaletteColors } from "./colors";
import { FONT } from "./fonts";

export default {
  title: "Theme/Fonts",
  parameters: { options: { showPanel: false } },
};

const codeStyles = {
  backgroundColor: lightPaletteColors.PRIMARY_3,
  padding: "2px 8px",
  marginBottom: "62px",
  fontSize: "14px",
  fontFamily: "Lucida Console, Courier, monospace",
} as React.CSSProperties;

const Item = ({
  itemKey,
  itemValue,
  fontType,
}: {
  itemKey: string;
  itemValue: string;
  fontType: string;
}) => {
  const renderStyles = {
    margin: 0,
    lineHeight: 1,
    fontFamily: fontType === "family" ? itemValue : FONT.family.title,
    fontSize: fontType === "size" ? itemValue : FONT.size.XXL,
    fontWeight: fontType === "weight" ? itemValue : FONT.weight.SEMI_BOLD,
  } as React.CSSProperties;

  return (
    <>
      <h3 style={renderStyles}>The quick brown fox jumps over the lazy dog</h3>
      <pre style={codeStyles}>
        Key: {itemKey} - Value: {itemValue}
      </pre>
    </>
  );
};

const mapFonts = (obj: object, fontType: string): React.ReactNode => {
  const fonts: React.ReactNodeArray = [];
  Object.entries(obj).forEach(([key, value]) =>
    fonts.push(
      <Item itemKey={key} itemValue={value} fontType={fontType} key={key} />
    )
  );

  return fonts;
};

export const Family = () => mapFonts(FONT.family, "family");

export const Size = () => mapFonts(FONT.size, "size");

export const Weight = () => mapFonts(FONT.weight, "weight");
