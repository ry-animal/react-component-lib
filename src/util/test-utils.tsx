import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
