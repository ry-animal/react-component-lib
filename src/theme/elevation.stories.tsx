import styled from "styled-components";
import { Text } from "../components/Text";
import { ELEVATION_LIGHT } from "./elevation";

const BoxContainer = styled.div`
  display: flex;
  padding: ${({ theme }) => theme.space.s24};
  flex-direction: row;
  margin-bottom: ${({ theme }) => theme.space.s8};
`;

interface BoxProps {
  item: string;
}
const Box = styled.div<BoxProps>`
  align-self: center;
  background-color: ${({ theme }) => theme.color.SURFACE_2};
  margin-right: ${({ theme }) => theme.space.s32};
  width: 300px;
  height: 300px;
  box-shadow: ${({ theme, item }) => {
    if (item === "low") {
      return theme.elevation.low;
    }
    if (item === "high") {
      return theme.elevation.high;
    }
    return theme.elevation.medium;
  }};
`;

export default {
  title: "Theme/Elevation",
  parameters: { options: { showPanel: false } },
};

const ElevationItem = ({ name, value }: { name: string; value: string }) => (
  <BoxContainer>
    <Box item={name} />
    <div>
      <Text>
        <b>Key:</b> {name}
      </Text>
      <Text>
        <b>Value:</b> {value}
      </Text>
    </div>
  </BoxContainer>
);

export const ElevationList = (): JSX.Element[] =>
  Object.entries(ELEVATION_LIGHT).map(([key, value]) => (
    <ElevationItem key={key} name={key} value={value} />
  ));
