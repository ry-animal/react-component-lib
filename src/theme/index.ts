import { getColorsByTheme } from "./colors";
import { ELEVATION_DARK, ELEVATION_LIGHT } from "./elevation";
import { FONT, FontFamily, FontSize, FontWeight } from "./fonts";
import { GRADIENT } from "./gradient";
import { LAYERS } from "./layers";
import { MEDIA_WIDTHS } from "./media";
import { OPACITY } from "./opacity";
import { RADIUS } from "./radius";
import { SPACE } from "./space";
import { ThemeName } from "./theme.interface";

const sharedThemeKeys = {
  radius: {
    ...RADIUS,
  },
  space: {
    ...SPACE,
  },
  type: {
    ...FONT,
  },
  layer: {
    ...LAYERS,
  },
  media: {
    ...MEDIA_WIDTHS,
  },
  opacity: {
    ...OPACITY,
  },
  gradient: {
    ...GRADIENT,
  },
};

export const lightColors = getColorsByTheme(ThemeName.LIGHT);
export const darkColors = getColorsByTheme(ThemeName.DARK);

export const LIGHT_THEME = {
  name: ThemeName.LIGHT,
  ...sharedThemeKeys,
  color: {
    ...lightColors,
  },
  elevation: {
    ...ELEVATION_LIGHT,
  },
};

export const DARK_THEME = {
  name: ThemeName.DARK,
  ...sharedThemeKeys,
  color: {
    ...darkColors,
  },
  elevation: {
    ...ELEVATION_DARK,
  },
};

// Legacy
export { FontFamily, FontSize, FontWeight, MEDIA_WIDTHS, SPACE };
