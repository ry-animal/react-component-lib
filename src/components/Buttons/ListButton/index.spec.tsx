import userEvent from "@testing-library/user-event";
import { ListButton } from ".";
import { darkPaletteBase } from "../../../theme/colors";
import { render, screen } from "../../../util/test-utils";

describe("<ListButton />", () => {
  const sampleProps = () => ({
    label: "Some Label",
    onClick: jest.fn(),
  });

  describe("rendering", () => {
    test("Default rendering matches snapshot", () => {
      expect(
        render(<ListButton {...sampleProps()} />).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (with description)", () => {
      expect(
        render(
          <ListButton {...sampleProps()} description="Description" />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (with icon)", () => {
      expect(
        render(
          <ListButton {...sampleProps()} icon={"https://test-image-url.com"} />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (with description and icon)", () => {
      expect(
        render(
          <ListButton
            {...sampleProps()}
            description="Description"
            icon={"https://test-image-url.com"}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (icon at opposite end)", () => {
      expect(
        render(
          <ListButton
            {...sampleProps()}
            icon="https://test-image-url.com"
            flipIcon
          />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (custom label color)", () => {
      expect(
        render(
          <ListButton {...sampleProps()} labelColor={darkPaletteBase.TEXT_2} />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (custom description color)", () => {
      expect(
        render(
          <ListButton
            {...sampleProps()}
            descriptionColor={darkPaletteBase.TEXT_3}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (custom button background color)", () => {
      expect(
        render(
          <ListButton
            {...sampleProps()}
            backgroundColor={darkPaletteBase.SURFACE_1}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Default rendering matches snapshot (disabled)", () => {
      expect(
        render(<ListButton {...sampleProps()} disabled={true} />).asFragment()
      ).toMatchSnapshot();
    });
  });

  describe("event handling", () => {
    test("onClick called", () => {
      const props = sampleProps();
      render(<ListButton {...props} />);
      const button = screen.getByRole("button");
      userEvent.click(button);
      expect(props.onClick).toHaveBeenCalled();
    });
  });
});
