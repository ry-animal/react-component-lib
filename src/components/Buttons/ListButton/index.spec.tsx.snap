// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListButton /> rendering Default rendering matches snapshot (custom button background color) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #151515;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #151515;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c4 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Some Label
        </span>
        <p
          class="c4"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (custom description color) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c4 {
  color: #DBDBDB;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Some Label
        </span>
        <p
          class="c4"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (custom label color) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #EEEEEE;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c4 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Some Label
        </span>
        <p
          class="c4"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (disabled) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c4 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
    disabled=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Some Label
        </span>
        <p
          class="c4"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (icon at opposite end) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c5 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c2 {
    height: 20px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Some Label-icon"
        class="c2"
        src="https://test-image-url.com"
      />
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          Some Label
        </span>
        <p
          class="c5"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (with description and icon) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c5 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c2 {
    height: 20px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Some Label-icon"
        class="c2"
        src="https://test-image-url.com"
      />
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          Some Label
        </span>
        <p
          class="c5"
        >
          Description
        </p>
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (with description) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c4 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Some Label
        </span>
        <p
          class="c4"
        >
          Description
        </p>
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot (with icon) 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c5 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c2 {
    height: 20px;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Some Label-icon"
        class="c2"
        src="https://test-image-url.com"
      />
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          Some Label
        </span>
        <p
          class="c5"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<ListButton /> rendering Default rendering matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  background-color: #EEEEEE;
  padding: 20px;
  border-radius: 8px;
  border: 1px solid transparent;
  width: 100%;
  cursor: pointer;
}

.c0:hover {
  background-color: #FFFFFF;
  border: 1px solid #A4A4A4;
}

.c0:disabled {
  background-color: transparent;
  opacity: 0.6;
  cursor: default;
  border: 1px solid #F9F9F9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  color: #404040;
  font-family: 'Poppins',sans-serif;
  font-size: 16px;
  font-weight: 700;
  text-align: left;
}

.c4 {
  color: #5F5F5F;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 12px;
  margin-top: 0;
  margin-bottom: 0;
  max-width: 280px;
  text-align: left;
}

@media (max-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

<button
    aria-label="Some Label"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Some Label
        </span>
        <p
          class="c4"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;
